src/GraphQL/Types/Relay/Cursor.cs

```diff
@@ -0,0 +1,70 @@
+namespace GraphQL.Types.Relay
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+
+    public static class Cursor
```

Comment:
This is a bit misleading, this is really an `IndexCursor`. While it's definitely helpful in simple cases, generally you probably don't actually want to use index cursors for serious implementations, as it doesn't account for ordering and can't be derived from just an item, which makes it inefficient if you are backing GQL with an ORM.

That is not say it's bad, but i worry that providing it as a default under the generic name "Cursor" suggests to much for it's use.

Reply:
Thats interesting, what is the more efficient alternative? Storing the cursor in the database, so it doesn't have to be recalculated? In my head I was thinking you could use an `orderby` field that determines which property on a model to use as the cursor and which property on the model to run an `ORDER BY` on if you are using SQL.

On a related efficiency note, I asked the question ["Does GraphQL negate the need for Graph Databases"](https://stackoverflow.com/questions/50134500/does-graphql-negate-the-need-for-graph-databases) on StackOverflow. For a personal project, I was thinking of using Microsoft Orleans because it's cheap to run but CosmosDB or some other graph database might also be interesting from an efficiency standpoint. Neo4j even has some GraphQL integration.

Reply:
> what is the more efficient alternative?

Building the cursor from an unique id, is the most straightforward, and doesn't create collisions. _Ideally_ any cursor should be derivable from just a single item, so you don't have to query the DB for _where_ that item is  total. Think about a `create` mutation, often you want to return the created node, as well as an `Edge<T>` for that node, in the mutation payload. It's generally not easy to efficiently insert the new item and also get its index. If you can derive it from the item tho, you don't have to do any additional lookups. The other thing is you don't collisions when fetching connections for multiple sort or filter situations.

Reply:
@jquense Take a look at my [GraphQL.NET project template](https://github.com/Dotnet-Boxed/Templates) or specifically [this](https://github.com/Dotnet-Boxed/Templates/blob/8e7c051b121237564189ec2a90763842000e7a38/Source/Content/GraphQLTemplate/Schemas/QueryObject.cs#L79) relay connection for droids I'm trying to add. It talks to a [droid repository](https://github.com/Dotnet-Boxed/Templates/blob/master/Source/Content/GraphQLTemplate/Repositories/DroidRepository.cs).

The connection is ordering droids by their created date (date of birth). If you use the ID for the cursor as opposed to the created date property, this makes it more difficult to get the data from the database. Currently in the droid repository I call:

```
public Task<List<Droid>> GetDroids(
    int? first,
    DateTime? createdAfter,
    CancellationToken cancellationToken) =>
    Task.FromResult(Database
        .Droids
        .If(createdAfter.HasValue, x => x.Where(y => y.Created > createdAfter.Value))
        .If(first.HasValue, x => x.Take(first.Value))
        .ToList());
```

If the droid created date is the cursor, I can simply pass that into the query above. If the ID is the cursor, I'm not sure what the most efficient way to write that query would be (assuming we are using EF Core or some other SQL/Linq based data store). Perhaps I would have to get the created date of the droid with the given ID and then use that in the query above. Am I missing something?